### PR TITLE
add flow name to sub flows on flow run graph

### PR DESCRIPTION
**Pull request by brandonreid**
Add the flow name along side the flow-run name for sub-flow-runs.

image

Related to PrefectHQ/prefect#8969

### CODE REVIEW

File: src/components/FlowRunTimeline.vue [hunk 1]
```diff
@@ -309,11 +309,45 @@
...
+
   const subFlowRunLabels = computed(() => {
     return (subFlowRunsSubscription.response ?? [])
       .reduce((acc, curr) => {
         if (curr.name) {
-          acc.set(curr.id, curr.name)
+          let subFlowRunName = ''
+
+          if (subFlowSubscription.response) {
+            const subFlow = subFlowSubscription.response.find((flow) => flow.id === curr.flowId)
```
Comment (pleek91) [inline src/components/FlowRunTimeline.vue:342]: Do you have an idea of how many subflows might exist? If there are a lot we probably should convert the response to a map keyed by the id so we can do lookups like this without looping.

File: src/components/FlowRunTimelineSubFlowRunDetails.vue [hunk 2]
```diff
@@ -43,18 +40,50 @@
...
-
   const api = useWorkspaceApi()
+
   const flowRunSubscriptionArgs = computed<[string]>(() => [flowRunId.value])
   const flowRunSubscription = useSubscriptionWithDependencies(
     api.flowRuns.getFlowRun,
     flowRunSubscriptionArgs,
   )
 
   const subFlowRun = computed(() => flowRunSubscription.response)
```
Comment (pleek91) [inline src/components/FlowRunTimelineSubFlowRunDetails.vue:51]: This subscription is more complex than it needs to be. You're just getting a single flow run right? 
```typescript
const { flowRunId } = toRefs(props)
const { flowRun: subFlowRun } = useFlowRun(flowRunId)
```

File: src/components/FlowRunTimelineSubFlowRunDetails.vue [hunk 3]
```diff
@@ -43,18 +40,50 @@
...
+
+  const flowSubscriptionArgs = computed<[string]>(() => [flowRunSubscription.response?.flowId || ''])
+  const flowSubscription = useSubscriptionWithDependencies(
+    api.flows.getFlow,
+    flowSubscriptionArgs,
+  )
+
+  const flow = computed(() => flowSubscription.response)
```
Comment (pleek91) [inline src/components/FlowRunTimelineSubFlowRunDetails.vue:59]: Same here
```
const flowId = computed(() => subFlowRun.value?.flowId)
const { flow } = useFlow(flowId)
```

File: src/components/FlowRunTimeline.vue [hunk 4]
```diff
@@ -309,11 +309,45 @@
...
+          id: subFlowIds,
+        },
+      },
+    ]
+  })
+
+  const subFlowSubscription = useSubscriptionWithDependencies(
+    api.flows.getFlows,
+    subFlowsSubscriptionFilter,
+  )
```
Comment (pleek91) [inline src/components/FlowRunTimeline.vue:333]: We should make a `useFlows` composition for this.

File: src/components/FlowRunTimelineSubFlowRunDetails.vue (hunk 3 above)
Comment (brandonreid) [reply src/components/FlowRunTimelineSubFlowRunDetails.vue:59]: Maybe we can look at this together when we pair. When another node is selected, the component doesn't remount, it get's a new ID. So here it'll actually update the data, but there's probably another way.

Comment (pleek91) [reply src/components/FlowRunTimelineSubFlowRunDetails.vue:59]: The id is reactive. So if flowId changes it'll fetch the new flow.

File: src/components/FlowRunTimelineSubFlowRunDetails.vue (hunk 2 above)
Comment (brandonreid) [reply src/components/FlowRunTimelineSubFlowRunDetails.vue:51]: ahhh, didn't know this existed

Comment (brandonreid) [reply src/components/FlowRunTimelineSubFlowRunDetails.vue]: Updated the task run details panel to use the composition as well.

File: src/components/FlowRunTimeline.vue (hunk 4 above)
Comment (brandonreid) [reply src/components/FlowRunTimeline.vue]: You [resolved](https://github.com/PrefectHQ/prefect-ui-library/pull/1307) this one for me!

File: src/components/FlowRunTimeline.vue [hunk 5]
```diff
@@ -292,6 +292,7 @@
     })
 
     expandedSubFlowRuns.value.set(id, {
+      // @ts-expect-error data actually is a valid property
```
Comment (pleek91) [inline src/components/FlowRunTimeline.vue:296]: Did this error not go away when you were not linked? There shouldn't be a typescript error.

Comment (brandonreid) [reply src/components/FlowRunTimeline.vue]: I forgot to `npm ci` after resetting `package.json`. Yeah it's not complaining now.

**Review by pleek91 — APPROVED** [review]

**Pull request merged**